api/server/router/container/container_routes.go

```diff
@@ -317,8 +319,26 @@ func (s *containerRouter) postContainersWait(ctx context.Context, w http.Respons
 	// Block on the result of the wait operation.
 	status := <-waitC
 
+	// With API < 1.34, wait on WaitConditionRemoved did not return
+	// in case container removal failed. Emulate it for older clients
+	// by repeating the wait until there is no error.
```

Comment:
Instead of continuing to wait, I think it would be better to just return the error as an internal error. This way, older clients will still unblock but handle it differently and return the error.

Reply:
I mean, this part of the API (error during wait) was unspecified in those older versions but it's better to unblock the client with an error than keep it blocking indefinitely while unaware that an error ever happened.

Reply:
But the client wouldn't get the error if its an older client and most likely expecting when unblocked that it can perform some other task that it may not be able to due to that error.
I just think this would be a breaking change for older clients that will cause issues in hard to discover ways.
A blocked client is much easier to debug and even potentially work-around.

Reply:
you're right 👍  @kolyshkin please disregard my previous comments.

Reply:
I agree with @jlhawn, I don't like the fact that we need to update both the daemon and the client for the bug to go away. The problem is, since api version 1.30 we return http 200 OK right away, and so there is no way to return any error later.

Hmm, now when I think about it, there is a way! We can return an error by closing the connection (same thing is happening in case of daemon shutdown), meaning we failed to wait for container removal. This way, docker-cli will be unstuck and print an error ("connection unexpectedly closed" or smth), and in general no client can think that the container is removed as JSON with StatusCode is not received. This is a kludge, too, but it makes sense.

@cpuguy83 what do you think? Oh and by the way, can you point me out to any other client than docker-cli (I'm a bit new to this and don't know of any).